SonarQube.Client/RoslynExporterAdapter/NuGetPackageInfoGenerator.cs

```diff
@@ -0,0 +1,59 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using SonarQube.Client.Messages;
+using SonarQube.Client.Models;
+
+namespace SonarQube.Client.RoslynExporterAdapter
+{
+    public static class NuGetPackageInfoGenerator
+    {
+        public static IEnumerable<NuGetPackageInfoResponse> GetNuGetPackageInfos(IEnumerable<SonarQubeRule> activeRules,
+            IDictionary<string, string> sonarProperties)
+        {
+            var propertyPrefixes = GetPluginPropertyPrefixes(activeRules);
+            var packages = new List<NuGetPackageInfoResponse>();
+
+            foreach (var partialRepoKey in propertyPrefixes)
+            {
+                if (!sonarProperties.TryGetValue($"{partialRepoKey}.analyzerId", out var analyzerId) ||
```

Comment:
If we reached this point -- is it possible that those properties are missing?
It assumes that we already have a rule with `roslyn.<something>`, is it possible to have a rule without those properties?

Reply:
It's a good practice to code defensively in extensibility scenarios i.e. where code we don't own is being plugged in. It's better to code defensively that crash at runtime because of what happens in third-party code.

However it shouldn't happen for plugins created using the SonarQube Roslyn SDK.